### PR TITLE
User/Group Management Dashboards

### PR DESCRIPTION
**Pull request by janniten**
User and Group Management Events Dashboards

Screenshot_2019-12-20  winlogbeat  User Management Events - Kibana
Screenshot_2019-12-20  winlogbeat  Group Management Events - Kibana

### CODE REVIEW

**Comment by janniten** [comment]
@leehinman I have created a new PR only for dashboards of user and group events

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by andrewkroh** [comment]
I fixed up the dashboards. The only issue with them was that the format was slightly different than what is expected for the dashboards that are checked into the repo. The ones in the repo get decoded to make diffing a little easier, then are encoded again when released in packages. I think there is a `--decode` flag than can be set to do this like `.\winlogbeat.exe export dashboard --decode --id=<uuid>`.

**Comment by andrewkroh** [comment]
Looks like we hit a few linter issues. I'll try to fix them:

```
>> check: Checking source code for common problems
>> Dashboard format - module/security/_meta/kibana/7/dashboard/01c54730-fee6-11e9-8405-516218e3d268.json:
   expected title with format '[Winlogbeat Module] Some title', found '[Winlogbeat] Group Management Events - Simple Metrics': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Description': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Event Actions - Donut': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Event Actions - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Groups vs Actions - Heatmap': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Created - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Changes - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Deleted - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Added - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Removed from Group - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found '5_Group_Enum': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Added - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Deleted- Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Created - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Changes - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Removed from Group  - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Membership Enumeration  - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Action Distribution over Time': title doesn't match pattern
>> Dashboard format - module/security/_meta/kibana/7/dashboard/71f720f0-ff18-11e9-8405-516218e3d268.json:
   expected title with format '[Winlogbeat Module] Some title', found '[Winlogbeat] User Management Events': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User  Management Events - Description': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Event Actions - Donut': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Created - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Event Actions - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Enabled - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Disabled - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Deleted - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Affected Users vs Actions - Heatmap': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Password Changes - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Unlocked Users - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Changes  Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Locked Out - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found '4_Event_Distribution in time': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Disabled - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Enabled - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Deleted - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Created - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Unlocks - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Password Changes - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users locked Out - VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Changes  VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Renamed VB Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Renamed - Table': title doesn't match pattern
>> Dashboard format - module/security/_meta/kibana/7/dashboard/8223bed0-b9e9-11e9-b6a2-c9b4015c4baf.json:
   expected title with format '[Winlogbeat Module] Some title', found '[Winlogbeat] User Management Events - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User  Management Events - Description': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Event Actions - Donut': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Created - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Event Actions - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Enabled - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Disabled - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Deleted - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'User Management Events - Affected Users vs Actions - Heatmap': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Password Changes - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Created - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Enabled - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Deleted - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Disabled - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Password Reset / Changes': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Unlocked Users - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Changes  Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Unlocks - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Unlocked Users - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Changes - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Locked Out - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found '4_Event_Distribution in time': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Renamed - Simple Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Renamed - Table': title doesn't match pattern
>> Dashboard format - module/security/_meta/kibana/7/dashboard/bb858830-f412-11e9-8405-516218e3d268.json:
   expected title with format '[Winlogbeat Module] Some title', found '[Winlogbeat] Group Management Events': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Description': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Event Actions - Donut': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Event Actions - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Added - Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Deleted - Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Created - Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Changed - Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Removed - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Enumeration - Metric': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Events - Groups vs Actions - Heatmap': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Created - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Changes - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Groups Deleted - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Added - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Users Removed from Group - Table': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found '5_Group_Enum': title doesn't match pattern
   expected title with format 'Some title [Winlogbeat Module]', found 'Group Management Action Distribution over Time': title doesn't match pattern
Error: there are format errors in dashboards
```

**Comment by andrewkroh** [comment]
jenkins, test this

File: x-pack/winlogbeat/module/security/_meta/kibana/7/dashboard/71f720f0-ff18-11e9-8405-516218e3d268.json [hunk 1]
```diff
@@ -0,0 +1,4399 @@
...
+              },
+              "schema": "bucket",
+              "type": "terms"
+            },
+            {
+              "enabled": true,
+              "id": "4",
+              "params": {
+                "customLabel": "Performer LogonID",
+                "field": "winlog.logon.id",
```
Comment (nicpenning) [inline x-pack/winlogbeat/module/security/_meta/kibana/7/dashboard/71f720f0-ff18-11e9-8405-516218e3d268.json:880]: It appears a user from the community (Using a fresh 7.8.0 stack) noticed that winlog.logon.id does not exist in the winlogbeat-* index pattern from the setup which yields in an error message "Could not locate that index-pattern-field (id: winlog.logon.id)" when trying to edit/look at some of the new visualizations such as "Users Deleted - Table [Winlogbeat Security]"

Not sure if I follow but it looks like the index pattern is "generated" from the fields.yml and if that is true I can see why this field does not get added to the winlogbeat-* index pattern.

The fields.common.yml is missing winlog.logon.id so maybe that is why the default index pattern that gets generated may not have this field?

Slack thread here: https://elasticstack.slack.com/archives/CNEDGGJQ3/p1595052755138500

Maybe this is a case of Dashboards/Visualizations using a custom index pattern that isn't fully compliant with the default winlogbeat-* index pattern created from the setup?

If users don't get all of the event logs that would contain these fields then the index pattern would never contain this field, so maybe it is worth adding into the fields.common.yml? I just wanted to make a note of this analysis.

Comment (janniten) [reply x-pack/winlogbeat/module/security/_meta/kibana/7/dashboard/71f720f0-ff18-11e9-8405-516218e3d268.json:880]: Hi @nicpenning,
I don't know exactly how the winlogbeat index pattern is generated, but probably I'll need to add the field winlog.logon.id in the security module specific fields.yml file (beats\x-pack\winlogbeat\module\security\_meta\fields.yml)
@andrewkroh, what do you think?  Should I add this field here or in any other place?
In addition. Which is the criteria in order to add a field in the yaml? The winlog.logon.id is not the only field I've added; I can add the necessary fields.
Thank you!
Regards

**Pull request merged**